### PR TITLE
Added Cart Management and some modifcations in context

### LINKED ISSUE #8: Add and Remove from cart for logged in user

### LINKED ISSUE #9: Increase and decrease the cart quantity of a particular product

### PR DESCRIPTION
**Pull request by romabulani**
- Renamed following files as I had to maintain one context and one reducer
dataContext => productContext
filterContext => dataContext
handleFilters => reducer
- Add to cart, increment, decrement, remove from cart
- Calculation of cart price dynamically
- Removed yarn.lock as its not needed

[Deploy preview](https://deploy-preview-10--affectionate-kepler-d776bc.netlify.app/)

### CODE REVIEW

**Review by anuj-kumary — APPROVED** [review]
- Very Clean code and UI. Loved the detailed description of the preview for PR.
- Good semantic HTML tags.
- Good naming conventions

**Suggestion in UI**
- When a user is not logged in and try to add a product to the cart so send the user to login page.

File: src/components/cart/CartItem.jsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    else {
+      if (item.qty === 1) response = await removeFromCart(item._id, authToken);
+      else response = await updateQuantityInCart(item._id, authToken, type);
+    }
+    dispatch({ type: "CART_OPERATION", payload: { cart: response.cart } });
+  };
+
+  const removeItem = async (item) => {
+    const response = await removeFromCart(item._id, authToken);
+    dispatch({ type: "CART_OPERATION", payload: { cart: response.cart } });
```
Comment (Rohanmond) [inline src/components/cart/CartItem.jsx:26]: CART_OPERATION can be inside some constant variable so you don't have to change throughout your app

File: src/components/profile/ProfileDetails.jsx [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 
 function ProfileDetails() {
   const { setAuthToken, authUser, setAuthUser } = useAuth();
   const navigate = useNavigate();
+  useEffect(() => {
+    if (localStorage.authUser) setAuthUser(JSON.parse(localStorage.authUser));
+  }, []);
   function logoutHandler() {
     localStorage.removeItem("authToken");
     localStorage.removeItem("authUser");
```
Comment (Rohanmond) [inline src/components/profile/ProfileDetails.jsx:14]: not sure but directly storing user data to local storage ...will it not concern privacy issues

**Review by Rohanmond — APPROVED** [review]

**Comment by mansipadgelwar** [comment]
Code is well structured. Great work Roma.

File: src/components/cart/cart.css [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  width: 100%;
+  height: 2rem;
+  margin: unset;
+  margin: 0.5rem 0;
+}
+
+.coupon-button:hover {
+  background-color: var(--gray-color);
+  color: unset;
+  /* width: 100%; */
```
Comment (samyakshah3008) [inline src/components/cart/cart.css]: Small suggestion: Can remove this commented code later on as refactoring

**Comment by romabulani** [comment]
> Code is well structured. Great work Roma.

Thankyou :)

File: src/contexts/dataContext.js [hunk 4]
```diff
@@ -1,30 +1,112 @@
...
+    productsData = funcSortHighToLow(productsData);
+  else if (state.sortBy === "lowToHigh")
+    productsData = funcSortLowToHigh(productsData);
+
+  if (!state.isOutOfStock) productsData = funcInStock(productsData);
+
+  // cakes and muffins
+
+  // getting keys with values true, keys are category names of items
+  const itemCategories = Object.keys(state.items).filter(
```
Comment (samyakshah3008) [inline src/contexts/dataContext.js:54]: Good to know that you have included these comments, it gives better code review

**Comment by romabulani** [comment]
> * Very Clean code and UI. Loved the detailed description of the preview for PR.
> * Good semantic HTML tags.
> * Good naming conventions
> 
> **Suggestion in UI**
> 
> * When a user is not logged in and try to add a product to the cart so send the user to login page.


Thankyou for pointing it out, have made the suggested changes

**Comment by samyakshah3008** [comment]
image

Can you check out why this image is not getting load in mine? is it for me only?

**Review by samyakshah3008 — APPROVED** [review]
I have included some suggestions, please see single comments 
Rest is all good for approval, loved your UI and code quality ✅

File: src/contexts/dataContext.js (hunk 4 above)
Comment (romabulani) [reply src/contexts/dataContext.js:54]: > image
> 
> Can you check out why this image is not getting load in mine? is it for me only?

Yes, the image is loading for me
image

File: src/components/cart/CartItem.jsx (hunk 1 above)
Comment (romabulani) [reply src/components/cart/CartItem.jsx:26]: Sure, will implement this for all constants in refactor branch

**Comment by romabulani** [comment]
> I have included some suggestions, please see single comments Rest is all good for approval, loved your UI and code quality ✅

Thankyou :)

File: src/components/cart/cart.css (hunk 3 above)
Comment (romabulani) [reply src/components/cart/cart.css]: removed the comment, thankyou for pointing it out

**Pull request merged**